src/axom/core/ArrayBase.hpp

```diff
@@ -405,27 +445,25 @@ class ArrayBase
   /*! \brief Returns a scalar reference given a full set of indices */
   AXOM_HOST_DEVICE SliceType<DIM> sliceImpl(const StackArray<IndexType, DIM>& idx)
   {
-    const IndexType baseIdx =
-      numerics::dot_product((const IndexType*)idx, m_strides.begin(), DIM);
+    const IndexType baseIdx = offset(idx);
     assert(inBounds(baseIdx));
-    return asDerived().data()[baseIdx];
+    return asDerived().data()[baseIdx * asDerived().spacing()];
   }
 
   /// \overload
   AXOM_HOST_DEVICE ConstSliceType<DIM> sliceImpl(
     const StackArray<IndexType, DIM>& idx) const
   {
-    const IndexType baseIdx =
-      numerics::dot_product((const IndexType*)idx, m_strides.begin(), DIM);
+    const IndexType baseIdx = offset(idx);
     assert(inBounds(baseIdx));
-    return asDerived().data()[baseIdx];
+    return asDerived().data()[baseIdx * asDerived().spacing()];
   }
   /// @}
 
 protected:
-  /// \brief The sizes (extents?) in each dimension
-  StackArray<IndexType, DIM> m_dims;
-  /// \brief The strides in each dimension
+  /// \brief The extent in each direction
+  StackArray<IndexType, DIM> m_shape;
+  /// \brief Logical strides in each direction
```

Comment:
I thought the comment was clearer with "in each dimension" rather than "in each direction".

Reply:
I'm trying to get away from the overloaded word "dimension", which have been used to mean size (2x3), direction (i, j or k) or (the self-referencing) dimension (as in 2 dimensions or 3 dimensions).

Reply:
Fortran uses the term "rank"